src/mesh/voxel_cloud.rs

```diff
@@ -760,269 +496,238 @@ impl VoxelCloud {
+    relative_voxel_to_absolute_voxel_coordinate(&relative, block_start)
 }
 
-/// Computes the voxel-space coordinates of a voxel containing the input
-/// point.
-fn cartesian_to_absolute_voxel_coords(
-    point: &Point3<f32>,
+/// Computes a voxel position in world space cartesian units from an index to
+/// the linear representation of the voxel block.
+fn one_dimensional_to_cartesian_coordinate(
+    one_dimensional_coordinate: usize,
+    block_start: &Point3<i32>,
+    block_dimensions: &Vector3<u32>,
     voxel_dimensions: &Vector3<f32>,
-) -> Point3<i32> {
-    assert!(
-        !approx::relative_eq!(voxel_dimensions.x, 0.0)
-            && !approx::relative_eq!(voxel_dimensions.y, 0.0)
-            && !approx::relative_eq!(voxel_dimensions.z, 0.0),
-        "Voxel dimensions can't be 0.0"
-    );
-    Point3::new(
-        (point.x / voxel_dimensions.x).round() as i32,
-        (point.y / voxel_dimensions.y).round() as i32,
-        (point.z / voxel_dimensions.z).round() as i32,
-    )
+) -> Point3<f32> {
+    let relative =
+        one_dimensional_to_relative_voxel_coordinate(one_dimensional_coordinate, block_dimensions);
+    relative_voxel_to_cartesian_coordinate(&relative, block_start, voxel_dimensions)
 }
 
-/// Computes the voxel-space coordinates of a voxel containing the
-/// input point
-#[allow(dead_code)]
-fn cartesian_to_relative_voxel_coords(
-    point: &Point3<f32>,
+/// Computes the center of a voxel in absolute voxel units from voxel
+/// coordinates relative to the voxel block start.
+fn relative_voxel_to_absolute_voxel_coordinate(
+    relative_coordinate: &Point3<i32>,
     block_start: &Point3<i32>,
-    voxel_dimensions: Vector3<f32>,
 ) -> Point3<i32> {
-    assert!(
-        !approx::relative_eq!(voxel_dimensions.x, 0.0)
-            && !approx::relative_eq!(voxel_dimensions.y, 0.0)
-            && !approx::relative_eq!(voxel_dimensions.z, 0.0),
-        "Voxel dimensions can't be 0.0"
-    );
-    Point3::new(
-        (point.x / voxel_dimensions.x).round() as i32 - block_start.x,
-        (point.y / voxel_dimensions.y).round() as i32 - block_start.y,
-        (point.z / voxel_dimensions.z).round() as i32 - block_start.z,
-    )
+    relative_coordinate + block_start.coords
 }
 
-/// Computes the center of a voxel in model-space coordinates from
-/// absolute voxel coordinates (relative to the voxel space origin).
-fn absolute_voxel_to_cartesian_coords(
-    absolute_coords: &Point3<i32>,
-    voxel_dimensions: Vector3<f32>,
+/// Computes the center of a voxel in worlds space cartesian units from voxel
+/// coordinates relative to the voxel block start.
+///
+/// # Panics
+///
+/// Panics if any of the voxel dimensions is equal or below zero.
+fn relative_voxel_to_cartesian_coordinate(
+    relative_coordinate: &Point3<i32>,
+    block_start: &Point3<i32>,
+    voxel_dimensions: &Vector3<f32>,
 ) -> Point3<f32> {
     assert!(
```

Comment:
My previous comment on this is not resolved. We should be checking for a positive number here instead of just non-zero. The doc comment says we are. But we aren't.